src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1682,5 +1686,227 @@ internal static ulong ConvertToUInt64Checked(double value)
             ThrowHelper.ThrowOverflowException();
             return 0;
         }
+
+        [StackTraceHidden]
+        internal static int DivideChecked(int dividend, int divisor)
+        {
+#if TARGET_32BIT
+            if ((uint)(divisor + 1) <= 1)
+            {
+                if (divisor == 0)
+                {
+                    ThrowHelper.ThrowDivideByZeroException();
+                }
+                else if (divisor == -1)
+                {
+                    if (dividend == int.MinValue)
+                    {
+                        ThrowHelper.ThrowOverflowException();
+                    }
+                    return -dividend;
+                }
+            }
+
+            return ActualDivisionInt(dividend, divisor);
+#else
+            return DivideChecked(dividend, divisor);
+#endif
+        }
+
+        [StackTraceHidden]
+        internal static int ModulusChecked(int dividend, int divisor)
+        {
+#if TARGET_32BIT
+            if ((uint)(divisor + 1) <= 1)
+            {
+                if (divisor == 0)
+                {
+                    ThrowHelper.ThrowDivideByZeroException();
+                }
+                else if (divisor == -1)
+                {
+                    if (dividend == int.MinValue)
+                    {
+                        ThrowHelper.ThrowOverflowException();
+                    }
+                    return 0;
+                }
+            }
+
+            return ActualModulusInt(dividend, divisor);
+#else
+            return ModulusChecked(dividend, divisor);
+#endif
+        }
+
+        [StackTraceHidden]
+        internal static uint DivideUnsigned(uint dividend, uint divisor)
```

Comment:
If you want to avoid dealing with overloading and included the type in the name, it should be the C# name to follow the managed naming conventions, e.g. `DivideInt32`, `DivideUInt32`, etc.